src/main/java/com/autotune/analyzer/performanceProfiles/PerformanceProfileValidation.java

```diff
@@ -80,6 +89,17 @@ public ValidationOutputData validate(PerformanceProfile performanceProfile) {
         return validatePerformanceProfileData(performanceProfile);
     }
 
+    /**
+     * Validates function variables
+     *
+     * @param metricProfile Metric Profile Object to be validated
+     * @return Returns the ValidationOutputData containing the response based on the validation
+     */
+    public ValidationOutputData validateMetricProfile(PerformanceProfile metricProfile) {
```

Comment:
remove this function
and use validateMetricProfileData directly

Reply:
currently `validateMetricProfile()` calls private method `validateMetricProfileData()` encapsulating the detailed validation logic, will it be okay to make it publicly accessible?

Reply:
yes